### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by sasha-dranuta**
[DEMO LINK](https://sasha-dranuta.github.io/react_list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -41,55 +42,122 @@ export function getReorderedGoods(
...
+  };
+
+  reset = () => {
+    this.setState({ isReversed: false, sortType: SortType.NONE });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const visibleGoods = getReorderedGoods(goodsFromServer, this.state);
+    const displayReset = sortType !== SortType.NONE || isReversed;
```
Comment (sergii-nosachenko) [inline src/App.tsx:77]:
```suggestion
    const shoudShowReset = sortType !== SortType.NONE || isReversed;
```

File: src/App.tsx [hunk 2]
```diff
@@ -41,55 +42,122 @@ export function getReorderedGoods(
...
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
```
Comment (sergii-nosachenko) [inline src/App.tsx:94]: Move this logic to `getReorderedGoods` function

File: src/App.tsx [hunk 3]
```diff
@@ -41,55 +42,122 @@ export function getReorderedGoods(
...
+              'is-warning',
+              {
+                'is-light': !isReversed,
+              },
+            )}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+          {displayReset && (
```
Comment (sergii-nosachenko) [inline src/App.tsx:140]: Add space before

File: src/App.tsx [hunk 4]
```diff
@@ -41,55 +42,122 @@ export function getReorderedGoods(
...
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {visibleGoods.map(good => (
+              <li data-cy="Good" key={good}>
+                {good}
+              </li>
+            ))}
+          </ul>
         </ul>
```
Comment (sergii-nosachenko) [inline src/App.tsx:159]: Create a separate list component

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (sasha-dranuta) [reply src/App.tsx:77]: Fixed

File: src/App.tsx (hunk 3 above)
Comment (sasha-dranuta) [reply src/App.tsx:140]: Fixed

File: src/App.tsx (hunk 4 above)
Comment (sasha-dranuta) [reply src/App.tsx:159]: Fixed

File: src/App.tsx [hunk 5]
```diff
@@ -38,58 +40,120 @@ export function getReorderedGoods(
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
```
Comment (Andr1yK) [inline src/App.tsx:41]: clean up

File: src/App.tsx [hunk 6]
```diff
@@ -38,58 +40,120 @@ export function getReorderedGoods(
...
+  sortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reset = () => {
+    this.setState({ isReversed: false, sortType: SortType.NONE });
```
Comment (Andr1yK) [inline src/App.tsx:89]:
```suggestion
    this.setState({
      isReversed: false,
      sortType: SortType.NONE,
    });
```

Comment (Andr1yK) [reply src/App.tsx:89]: Writing each property of an object on a new line is a style convention in JavaScript that makes the code easier to read and maintain. Here are a few reasons why it's a good practice to follow:

- **Readability**: By writing each property on a new line, it makes the code more visually separated and easier to read. When an object has multiple properties, putting them all on a single line can make the code look cluttered and hard to follow.

- **Ease of modification**: When an object has multiple properties and they are all on a single line, it can be difficult to modify the object later on. If you want to add or remove a property, you will need to modify the entire line, which can be error-prone and time-consuming. By writing each property on a new line, you can easily modify, add or remove properties without affecting the rest of the object.

- **Consistency**: Following a consistent style throughout the codebase makes it easier for developers to read and understand the code. If every object in the codebase is written with each property on a new line, it creates a standard that makes it easier for other developers to understand the code.

So, to sum up, while writing all properties of an object on a single line is technically valid in JavaScript, writing each property on a new line is a convention that improves the readability, ease of modification, and consistency of the code.

File: src/App.tsx [hunk 7]
```diff
@@ -38,58 +40,120 @@ export function getReorderedGoods(
...
+              className="button is-danger is-light"
+              onClick={this.reset}
+            >
+              Reset
+            </button>
+          )}
+        </div>
 
-      <ul>
         <ul>
```
Comment (Andr1yK) [inline src/App.tsx:153]: why do we need ul here?

File: src/components/GoodsList.tsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+type Props = {
+  goods: string[],
+};
+
+export const GoodsList: React.FC<Props> = ({ goods }) => {
+  return (
```
Comment (Andr1yK) [inline src/components/GoodsList.tsx:8]: do we need return here?

**Review by Andr1yK — CHANGES_REQUESTED** [review]

**Review by sergii-nosachenko — APPROVED** [review]
